pkg/install/util.go

```diff
@@ -40,3 +42,39 @@ func ObjErrorf(obj kates.Object, format string, args ...interface{}) error {
 		obj.GetObjectKind().GroupVersionKind().Kind, obj.GetName(), obj.GetNamespace(),
 		fmt.Errorf(format, args...))
 }
+
+// AlreadyUndone means that an install action has already been undone, perhaps by manual user action
+type AlreadyUndone struct {
+	err error
+	msg string
+}
+
+func (e *AlreadyUndone) Error() string {
+	return fmt.Sprintf("%s: %v", e.msg, e.err)
+}
+
+func (e *AlreadyUndone) Unwrap() error {
+	return e.err
+}
+
+func NewAlreadyUndone(err error, msg string) error {
+	return &AlreadyUndone{err, msg}
+}
+
+// IsAlreadyUndone returns whether the given error -- possibly a multierror -- indicates that all actions have been undone.
+func IsAlreadyUndone(err error) bool {
+	var undone *AlreadyUndone
+	if errors.As(err, &undone) {
+		return true
+	}
+	var multi *multierror.Error
+	if !errors.As(err, &multi) {
+		return false
+	}
+	for _, err := range multi.WrappedErrors() {
+		if !errors.As(err, &undone) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Looking at `go-multierror`, I'm like 90% sure that this is unnecessary and that you can just use `errors.As(err, &AlreadyUndone{})`

Reply:
I don't think so -- this is slightly different. My understanding of `go-multierror` is that `errors.As` will return "is _any_ error in this chain an AlreadyUndone", whereas here we want _all_ errors in the chain to eventually resolve back to an `AlreadyUndone`